docs: Clarify override behavior of resourceLabels directive (#7287)
nextflow-io/nextflow: docs/reference/process.mdx

Resource labels are attached to underlying resources such as cloud VMs, and are intended for operational purposes such as cost tracking. They are not recorded in lineage metadata.

> When `resourceLabels` is specified multiple times in the config, only the last setting is used. Additionally, when `resourceLabels` is specified both in the config and the process definition, only the process definition is used.

Reviewer: ```suggestion
If `resourceLabels` is set multiple times in the config, Nextflow uses only the last setting. If `resourceLabels` is set in both the config and the process definition, Nextflow uses the process definition.
```
